GCEWindowsAgent/addresses.go

```diff
@@ -62,6 +65,152 @@ func (a *addressMgr) parseWSFCEnable() bool {
 	return false
 }
 
+func getForwardsFromRegistry(mac string) ([]string, error) {
+	regFwdIPs, err := readRegMultiString(addressKey, mac)
+	if err == errRegNotExist {
+		// The old agent stored MAC addresses without the ':',
+		// check for those and clean them up.
+		oldName := strings.Replace(mac, ":", "", -1)
+		regFwdIPs, err = readRegMultiString(addressKey, oldName)
+		if err == nil {
+			deleteRegKey(addressKey, oldName)
+		}
+	} else if err != nil {
+		return nil, err
+	}
+	return regFwdIPs, nil
+}
+
+func compareIPs(configuredIPs, desiredIPs []string) (toAdd, toRm []string) {
+	for _, desiredIP := range desiredIPs {
+		if !containsString(desiredIP, configuredIPs) {
+			toAdd = append(toAdd, desiredIP)
+		}
+	}
+
+	for _, configuredIP := range configuredIPs {
+		if !containsString(configuredIP, desiredIPs) {
+			toRm = append(toRm, configuredIP)
+		}
+	}
+	return toAdd, toRm
+}
+
+var badMAC []string
+
+func getInterfaceByMAC(mac string) (net.Interface, error) {
+	hwaddr, err := net.ParseMAC(mac)
+	if err != nil {
+		return net.Interface{}, err
+	}
+
+	ifs, err := net.Interfaces()
+	if err != nil {
+		return net.Interface{}, err
+	}
+
+	for _, iface := range ifs {
+		if iface.HardwareAddr.String() == hwaddr.String() {
+			return iface, nil
+		}
+	}
+	return net.Interface{}, fmt.Errorf("No interface found with MAC %s", mac)
+}
+
+func getRoutes(ifname string) ([]string, error) {
+	args := fmt.Sprintf("route list table local type local scope host dev %s proto %d", ifname, protoID)
```

Comment:
I don't think this reads much better than []string{"route", "list", "table", "local", ...}
and you save the parsing split call

Reply:
the line would be:
```
args := []string{"route", "add", "to", "local", ip, "scope", "host", "dev", ifname, "proto", strconv.Itoa(protoID)}
```
that's not very readable to me. the fact that three of the words aren't string constants isn't that easy to pick out compared to a formatted string.